improve documentation

wassupluke opened the pull request:
For #149 .

Wikis on github are weird. Not sure if this is the most correct way to do it but I enabled the wiki in the repo settings, wrote all the wiki pages, then realized github handles wikis separate from repos so I couldn't do a pull request.

Had to end up running the following in the root directory of the fork.
```bash
git clone https://github.com/wassupluke/Launcher.wiki.git
cp -r Launcher.wiki/ launcher.wiki
rm -rf Launcher.wiki/
git submodule add https://github.com/wassupluke/Launcher.wiki.git launcher.wiki
```

If this doesn't pull through properly on your end let me know and maybe just enable wikis for your repo in Settings>Features>Wikis. From there I could just copy/paste my wiki page content and update the pages as needed when things change.

wassupluke commented:
Let me preface this with: this is my first time setting up a GitHub wiki, that's how little experience I have with them.

I honestly don't think there is any advantage of using the GitHub wiki. Like you said, it's really just markdown files that it displays, has a little sidebar showing the filenames and the markdown header structure of each file.

I like the idea of being able to update the code and documentation together and keeping those updates linked with commits. That's beautiful. I was really hoping the building GitHub wiki would let us do that as well, but after building it, it doesn't look like that's the case.

The static site generator stuff I've done before, but only out of a dedicated repo (<username>.github.io) and don't currently know how to make a site build from a specific directory within a repo, so that'll be something to figure out.

If you would like, I can scrap this PR and open a new one with a simple directory containing my markdown files.

jrpie commented:
> The static site generator stuff I've done before, but only out of a dedicated repo (.github.io) and don't currently know how to make a site build from a specific directory within a repo, so that'll be something to figure out.

I think I'd want to host that on my server (or at least use a domain I own) to avoid even more dependence on github. Hugo [Book](https://themes.gohugo.io/themes/hugo-book/) looks quite clean imo.

(I'm not too happy about this project being hosted on GitHub in the first place. The advantage is that people already have accounts to contribute, but perhaps I should move to codeberg or even my on forgejo. I already have a mirror at [git.jrpie.de](https://git.jrpie.de/jrpie/launcher) since some people complained about the source code not being available in their countries).


> If you would like, I can scrap this PR and open a new one with a simple directory containing my markdown files.

That would be great 👍🏼  You can just put the files in `docs/` which I created for that very purpose.
Markdown is a very nice format and writing everything down is the main work anyway.

wassupluke commented:
> Hugo [Book](https://themes.gohugo.io/themes/hugo-book/) looks quite clean imo
Looks amazing <3

> You can just put the files in docs/
Done :+1:

Codacy is complaining about some of the markdown styling so I'll clean that up. Otherwise all the wiki docs are now in the docs directory and I squashed the two variants of `build.md` into a single document and renamed the file with changes from Finn's launcher for clarity.
